source/velocity_boundary_conditions/interface.cc

```diff
@@ -70,7 +70,8 @@ namespace aspect
     }
 
 
-
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
```

Comment:
I am worried that this is not supported by all compilers (intel for example). I know that we have a compile check for it inside deal.II. Can you do the same by using DEAL_II_DISABLE_EXTRA_DIAGNOSTICS and DEAL_II_ENABLE_EXTRA_DIAGNOSTICS?